### ISSUE TITLE
Add ability to specify dock layout, independent of the tray layout

### ISSUE THREAD

**Issue by codesardine** [joewing/jwm#441]
doing this will provide the ability to design more flexible layouts

```
<Tray layout="vertical"
<Dock layout="horizontal">
</Tray> 
```

**Comment by technosaurus** [comment]
If that is added, it would probably also need "wrap-width" and "wrap-height" to keep icons from getting so squished, that they become unrecognizable.

Legend:   [&] =  icon, Bottom is a simulated "Dock"

Wrap Example
|-----------------|
|[&] Menu         |
|[&] Show Desk|
|===========|
|[&][&][&][&][&]
[&][&][&][&][&]|

No Wrap
|-----------------|
|[&] Menu         |
|[&] Show Desk|
|===========|
|[&][&][&][&][&][&][&][&][&][&][&]|

These icons would either get squished or the tray would grow to cover to much of the screen.

**Comment by codesardine** [comment]
I agree maybe even the ability for the icons to slide under other icons when they run out of tray space horizontally, kind of like a grid
